pennylane/_qubit_device.py

```diff
@@ -179,12 +200,25 @@ def execute(self, circuit, **kwargs):
         # expvals and vars in superfluous arrays
         all_sampled = all(obs.return_type is Sample for obs in circuit.observables)
         if circuit.is_sampled and not all_sampled:
-            return self._asarray(results, dtype="object")
+            results = self._asarray(results, dtype="object")
+        else:
+            results = self._asarray(results)
+
+        if self._cache and circuit_hash not in self._cache_execute:
+            self._cache_execute[circuit_hash] = results
+            if len(self._cache_execute) > self._cache:
```

Comment:
Two thoughts came to mind:
* For memory-intensive simulations (e.g. >25 qubits, returning the statevector) it can be desirable at times recycle the cache. Perhaps worth adding this feature at some point?
* Tying in there, maybe nice to have `len(self._cache_execute)` as a property such that the current cache size can be checked?

Maybe these would come later if the need arises.

Reply:
Thanks @antalszava! What do you mean by recycling the cache?

Reply:
Posing a max size for the memory used by the cache (e.g., checking the cache object using `sys.getsizeof`) and if the limit is exceeded then deleting the cached results. Multiple consecutive simulations with large systems might result in running out of memory due to the cache. Might not be urgent, but could be nice to avoid down the line.